### PR TITLE
Applied clang-tidy modernize-loop-convert fixes for Framework/ 

### PR DESCRIPTION
**Pull request by edwardb96**
1. Applied automatic fixes which convert c and iterator style for loops to range-based for loops where appropriate in Framework/ code.
2. Fixed bad variable names chosen by clang-tidy for non-trivial plurals e.g `boxes` as a singular is not `boxe`.

**To test:**
1. Ensure builds pass.
2. Quick code inspection.

Refs #22183. 

**Release Notes** 
*Does not need to be in the release notes.*

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)?
- [ ] Are the unit tests small and test the class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] Do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/DataHandling/src/LoadSpiceXML2DDet.cpp [hunk 1]
```diff
@@ -669,20 +669,17 @@ MatrixWorkspace_sptr LoadSpiceXML2DDet::createMatrixWorkspaceVersion2(
   }     // END-FOR (xml nodes)
 
   // Add the property to output workspace
-  for (std::map<std::string, std::string>::iterator miter = str_log_map.begin();
-       miter != str_log_map.end(); ++miter) {
+  for (auto &miter : str_log_map) {
```
Comment (josephframsay) [inline Framework/DataHandling/src/LoadSpiceXML2DDet.cpp]: Could `miter` be replaced with something like `log_entry`? Similarly for line 676 and 680

File: Framework/DataHandling/test/LoadDetectorInfoTest.h [hunk 2]
```diff
@@ -301,8 +301,8 @@ class LoadDetectorInfoTest : public CxxTest::TestSuite {
 
     // read the parameters from some random detectors, they're parameters are
     // all set to the same thing
-    for (int i = 0; i < NUMRANDOM; ++i) {
-      const size_t detIndex = detInfo.indexOf(DETECTS[i]);
+    for (int i : DETECTS) {
```
Comment (josephframsay) [inline Framework/DataHandling/test/LoadDetectorInfoTest.h]: Could `i` be replaced with something like `detector`?

File: Framework/DataHandling/test/LoadEventPreNexus2Test.h [hunk 3]
```diff
@@ -135,8 +135,8 @@ class LoadEventPreNexus2Test : public CxxTest::TestSuite {
     Types::Core::DateAndTime start = it->first;
 
     std::vector<TofEvent> events1 = ew->getSpectrum(1000).getEvents();
-    for (size_t i = 0; i < events1.size(); i++) {
-      std::cout << (events1[i].pulseTime() - start) << " sec \n";
+    for (auto &i : events1) {
```
Comment (josephframsay) [inline Framework/DataHandling/test/LoadEventPreNexus2Test.h]: could `i` be replaced with `event`?

File: Framework/Geometry/test/IndexingUtilsTest.h [hunk 4]
```diff
@@ -232,8 +232,8 @@ class IndexingUtilsTest : public CxxTest::TestSuite {
   void test_Optimize_Direction() {
     std::vector<int> index_values;
     int correct_indices[] = {1, 4, 2, 0, 1, 3, 0, -1, 0, -1, -2, -3};
-    for (size_t i = 0; i < 12; i++) {
-      index_values.push_back(correct_indices[i]);
+    for (int correct_indice : correct_indices) {
```
Comment (josephframsay) [inline Framework/Geometry/test/IndexingUtilsTest.h]: `correct_index`?

File: Framework/Geometry/test/IndexingUtilsTest.h [hunk 5]
```diff
@@ -372,8 +372,8 @@ class IndexingUtilsTest : public CxxTest::TestSuite {
                             {2.66668320, 5.29605670, 7.9653444}};
 
     std::vector<V3D> directions;
-    for (size_t i = 0; i < 5; i++)
-      directions.emplace_back(vectors[i][0], vectors[i][1], vectors[i][2]);
+    for (auto &vector : vectors)
```
Comment (josephframsay) [inline Framework/Geometry/test/IndexingUtilsTest.h]: `vector` seems like a risky variable name - could this be something like `direction` instead? The same goes for line 413

File: Framework/Geometry/test/IndexingUtilsTest.h [hunk 6]
```diff
@@ -756,8 +756,8 @@ class IndexingUtilsTest : public CxxTest::TestSuite {
     TS_ASSERT_DELTA(direction_list[7].Z(), -0.211325, 1e-5);
 
     double dot_prod;
-    for (size_t i = 0; i < direction_list.size(); i++) {
-      dot_prod = axis.scalar_prod(direction_list[i]);
+    for (const auto &i : direction_list) {
```
Comment (josephframsay) [inline Framework/Geometry/test/IndexingUtilsTest.h]: could `i` be replaced with `direction`?

File: Framework/Kernel/test/PropertyManagerTest.h [hunk 7]
```diff
@@ -90,12 +90,12 @@ class PropertyManagerTest : public CxxTest::TestSuite {
     TS_ASSERT_EQUALS(manager.propertyCount(), 3);
 
     const std::vector<Property *> &props = manager.getProperties();
-    for (auto iter = props.begin(); iter != props.end(); ++iter) {
-      Property *prop = *iter;
+    for (auto iter : props) {
```
Comment (josephframsay) [inline Framework/Kernel/test/PropertyManagerTest.h]: `prop`?

File: Framework/LiveData/test/LoadLiveDataTest.h [hunk 8]
```diff
@@ -180,8 +180,8 @@ class LoadLiveDataTest : public CxxTest::TestSuite {
     TS_ASSERT_EQUALS(ws1->getNumberHistograms(), 2);
     double total;
     total = 0;
-    for (auto it = ws1->readY(0).begin(); it != ws1->readY(0).end(); it++)
-      total += *it;
+    for (double it : ws1->readY(0))
```
Comment (josephframsay) [inline Framework/LiveData/test/LoadLiveDataTest.h]: Could `it` be renamed something like `yValue`? Also could 184 be put inside braces? The same goes for line 194

File: Framework/MDAlgorithms/test/LoadMDTest.h [hunk 9]
```diff
@@ -253,8 +253,8 @@ class LoadMDTest : public CxxTest::TestSuite {
 
       typename std::vector<API::IMDNode *> boxes;
       ws->getBox()->getBoxes(boxes, 1000, false);
-      for (size_t i = 0; i < boxes.size(); i++) {
-        MDBox<MDE, nd> *box = dynamic_cast<MDBox<MDE, nd> *>(boxes[i]);
+      for (auto &boxe : boxes) {
```
Comment (josephframsay) [inline Framework/MDAlgorithms/test/LoadMDTest.h]: `box`?

File: Framework/Algorithms/test/DetectorEfficiencyVariationTest.h [hunk 10]
```diff
@@ -130,8 +130,8 @@ class DetectorEfficiencyVariationTest : public CxxTest::TestSuite {
       inputB->setBinEdges(j, x);
       std::vector<double> forInputA, forInputB;
       // the spectravalues will be multiples of the random numbers above
-      for (int l = 0; l < ySize; ++l) {
-        forInputA.push_back(yArray[l]);
+      for (double l : yArray) {
```
Comment (josephframsay) [inline Framework/Algorithms/test/DetectorEfficiencyVariationTest.h]: could `l` be renamed to `y`?

File: Framework/Algorithms/test/MaskBinsFromTableTest.h [hunk 11]
```diff
@@ -183,8 +183,8 @@ class MaskBinsFromTableTest : public CxxTest::TestSuite {
     speclist.push_back(6);
     speclist.push_back(7);
     speclist.push_back(8);
-    for (size_t iws = 0; iws < speclist.size(); ++iws) {
-      auto &yvec = WS->y(speclist[iws]);
+    for (int iws : speclist) {
```
Comment (josephframsay) [inline Framework/Algorithms/test/MaskBinsFromTableTest.h]: could `iws` be changed to something like `spectrum`?

File: Framework/Algorithms/test/SassenaFFTTest.h [hunk 12]
```diff
@@ -144,10 +144,10 @@ class SassenaFFTTest : public CxxTest::TestSuite {
       double sum = 0.0;
       double average = 0.0;
       MantidVec::iterator itx = xv.begin();
-      for (MantidVec::iterator it = yv.begin(); it != yv.end(); ++it) {
+      for (double &it : yv) {
```
Comment (josephframsay) [inline Framework/Algorithms/test/SassenaFFTTest.h]: could `it` be renamed to something like `y`? The same goes for line 189

File: Framework/CurveFitting/src/FuncMinimizers/TrustRegionMinimizer.cpp [hunk 13]
```diff
@@ -1274,10 +1274,9 @@ void solveSubproblem(int n, double radius, double f,
   inform.obj *= pow(scale_c, 2) / scale_h;
   inform.multiplier *= scale_h;
   inform.pole *= scale_h;
-  for (size_t i = 0; i < inform.history.size();
-       ++i) { //      do i = 1, inform.len_history
-    inform.history[i].lambda *= scale_h;
-    inform.history[i].x_norm *= scale_c / scale_h;
+  for (auto &i : inform.history) { //      do i = 1, inform.len_history
```
Comment (josephframsay) [inline Framework/CurveFitting/src/FuncMinimizers/TrustRegionMinimizer.cpp]: Could `i` be renamed something like `historyItem`?

File: Framework/CurveFitting/test/Algorithms/LeBailFunctionTest.h [hunk 14]
```diff
@@ -515,10 +515,10 @@ class LeBailFunctionTest : public CxxTest::TestSuite {
         1.950190,    1.613562,    1.335208,    1.104734,    0.914043,
         0.756362,    0.000000};
 
-    for (size_t i = 0; i < vecY.size(); ++i) {
+    for (double i : vecY) {
```
Comment (josephframsay) [inline Framework/CurveFitting/test/Algorithms/LeBailFunctionTest.h]: Could `i` be renamed `y`? Or maybe `yValue`?

File: Framework/DataHandling/test/LoadDetectorInfoTest.h (hunk 2 above)
Comment (josephframsay) [inline Framework/DataHandling/test/LoadDetectorInfoTest.h]: Could `i` be renamed `detector`?

File: Framework/DataHandling/test/LoadEventPreNexus2Test.h (hunk 3 above)
Comment (josephframsay) [inline Framework/DataHandling/test/LoadEventPreNexus2Test.h]: `event`?

File: Framework/Geometry/test/IndexingUtilsTest.h (hunk 4 above)
Comment (josephframsay) [inline Framework/Geometry/test/IndexingUtilsTest.h]: `correct_index`?

File: Framework/Geometry/test/IndexingUtilsTest.h (hunk 5 above)
Comment (josephframsay) [inline Framework/Geometry/test/IndexingUtilsTest.h]: `vector` seems a risky choice - could this be something like `direction` instead? ditto 413 and 759

File: Framework/Kernel/test/PropertyManagerTest.h (hunk 7 above)
Comment (josephframsay) [inline Framework/Kernel/test/PropertyManagerTest.h]: `prop`?

File: Framework/LiveData/test/LoadLiveDataTest.h (hunk 8 above)
Comment (josephframsay) [inline Framework/LiveData/test/LoadLiveDataTest.h]: could `it` be renamed `y`? and 184 be put in curly braces? ditto 194

File: Framework/MDAlgorithms/test/LoadMDTest.h (hunk 9 above)
Comment (josephframsay) [inline Framework/MDAlgorithms/test/LoadMDTest.h]: `box`?

**Review by josephframsay — CHANGES_REQUESTED** [review]
These are nice changes - it's cool that clang-tidy can do this for you! Unfortunately it seems to have picked unhelpfully generic names in a lot of cases - apologies for the super picky review comments! I also notice that not many of the loop variables get `const` on them - is this something that clang-tidy tries to pick up?

File: Framework/Kernel/test/PropertyManagerTest.h (hunk 7 above)
Comment (edwardb96) [reply Framework/Kernel/test/PropertyManagerTest.h]: Collides with the variable definition below.

File: Framework/Geometry/test/IndexingUtilsTest.h (hunk 5 above)
Comment (edwardb96) [reply Framework/Geometry/test/IndexingUtilsTest.h]: Could be confused as being an item in the `directions` vector on l 374. `vector` is accurate and isn't really 'risky' in this case.

File: Framework/Algorithms/test/RebinTest.h [hunk 15]
```diff
@@ -424,9 +424,8 @@ class RebinTest : public CxxTest::TestSuite {
 
     // turn the mask list into an array like the Y values
     MantidVec weights(outY.size(), 0);
-    for (MatrixWorkspace::MaskList::const_iterator it = mask.begin();
-         it != mask.end(); ++it) {
-      weights[it->first] = it->second;
+    for (auto it : mask) {
```
[the remark this answers is not in the document]
Comment (edwardb96) [reply Framework/Algorithms/test/RebinTest.h]: There probably is...

**Review by josephframsay — APPROVED** [review]

**Pull request merged**